string direction
uint8 speed_level